protocol/messenger_activity_center.go

```diff
@@ -0,0 +1,76 @@
+package protocol
+
+import (
+	"github.com/status-im/status-go/eth-node/types"
+)
+
+func (m *Messenger) UnreadActivityCenterNotificationsCount() (uint64, error) {
+	return m.persistence.UnreadActivityCenterNotificationsCount()
+}
+
+func (m *Messenger) MarkAllActivityCenterNotificationsRead() error {
+	return m.persistence.MarkAllActivityCenterNotificationsRead()
+}
+
+func (m *Messenger) processAcceptedActivityCenterNotifications(notifications []*ActivityCenterNotification) (*MessengerResponse, error) {
+	response := &MessengerResponse{}
+	m.mutex.Lock()
+	var chats []*Chat
+	for _, notification := range notifications {
+		if notification.ChatID != "" {
+			chat, ok := m.allChats.Load(notification.ChatID)
+			if !ok {
+				// This should not really happen, but ignore just in case it was deleted in the meantime
+				m.logger.Warn("chat not found")
+				continue
+			}
+			chat.Active = true
+
+			chats = append(chats, chat)
+			response.AddChat(chat)
+		}
+	}
+	if len(chats) != 0 {
+		err := m.saveChats(chats)
+		if err != nil {
+			return nil, err
+		}
+	}
+	m.mutex.Unlock()
+	return response, nil
+}
```

Comment:
You've got early returns in this function what would make this unlock unreachable.

```suggestion
func (m *Messenger) processAcceptedActivityCenterNotifications(notifications []*ActivityCenterNotification) (*MessengerResponse, error) {
	m.mutex.Lock()
	defer m.mutex.Unlock()
	
	response := &MessengerResponse{}
	var chats []*Chat
	for _, notification := range notifications {
		if notification.ChatID != "" {
			chat, ok := m.allChats.Load(notification.ChatID)
			if !ok {
				// This should not really happen, but ignore just in case it was deleted in the meantime
				m.logger.Warn("chat not found")
				continue
			}
			chat.Active = true

			chats = append(chats, chat)
			response.AddChat(chat)
		}
	}
	if len(chats) != 0 {
		err := m.saveChats(chats)
		if err != nil {
			return nil, err
		}
	}
	return response, nil
}
```

Reply:
nice catch!!!